src/ParallelAlgorithms/LinearSolver/ConjugateGradient/ElementActions.hpp

```diff
@@ -165,11 +157,6 @@ struct PerformStep {
     using operator_tag =
         db::add_tag_prefix<LinearSolver::Tags::OperatorAppliedTo, operand_tag>;
 
-    ASSERT(get<LinearSolver::Tags::IterationId<OptionsGroup>>(box) !=
```

Comment:
No, in fact `PrepareStep` doesn't currently do anything so nothing observable changes (it will do something in the preconditioned GMRES though, which is an upcoming PR).
The linear solvers currently rely on the actions that are exposed in the public-facing `Gmres` or `ConjugateGradient` are run in the correct order. I am considering to expose typelists where the "compute operator" actions are injected at the right places, but I'm not sure if that's cleaner and it takes away some of the freedom the executable has to run actions in-between.